geetools/Image/__init__.py

```diff
@@ -500,16 +502,30 @@ def fullLike(
                 image = image.geetools.fullLike(0)
                 print(image.bandNames().getInfo())
         """
+        # function params as GEE objects
         keepMask, copyProperties = ee.Number(keepMask), ee.Number(copyProperties)
+        keepFootprint = ee.Number(keepFootprint)
+        # get geometry, band names and property names
         footprint, bandNames = self._obj.geometry(), self._obj.bandNames()
+        properties = self._obj.propertyNames().remove(
+            "system:footprint"
+        )  # remove footprint as a "normal" property
+        # list of values to fill the image
         fillValue = ee.List.repeat(fillValue, bandNames.size())
-        image = (
-            self.full(fillValue, bandNames)
-            .reproject(self._obj.select(0).projection())
-            .clip(footprint)
+        # filled image
+        image = self.full(fillValue, bandNames)
+        # handler projection
+        projected_list = bandNames.map(
+            lambda b: image.select([b]).reproject(self._obj.select([b]).projection())
```

Comment:
I think you don't need to use the `[` in select (https://developers.google.com/earth-engine/apidocs/ee-image-select).